internal/ifelse/rule.go

```diff
@@ -47,59 +50,90 @@ type visitor struct {
 }
 
 func (v *visitor) Visit(node ast.Node) ast.Visitor {
-	block, ok := node.(*ast.BlockStmt)
-	if !ok {
+	switch stmt := node.(type) {
+	case *ast.FuncDecl:
+		if stmt.Body != nil {
+			v.visitBlock(stmt.Body.List, Return)
+		}
+	case *ast.FuncLit:
+		v.visitBlock(stmt.Body.List, Return)
+	case *ast.ForStmt:
+		v.visitBlock(stmt.Body.List, Continue)
+	case *ast.RangeStmt:
+		v.visitBlock(stmt.Body.List, Continue)
```

Comment:
Why would body be nil, only for the first one?

I would expect body nil to be tested everywhere

Reply:
The doc for `FuncDecl` mentions a case where it is `nil`:

> Body *[BlockStmt](https://pkg.go.dev/go/ast#BlockStmt)    // function body; or nil for external (non-Go) function

For the others, my understanding was they are always non-nil, but happy to be corrected. I guess it doesn't hurt to check either way.